### ISSUE TITLE
Ubuntu device type is 'Unknown' (instead of 'Computer')

### ISSUE THREAD

**Issue by itaykor** [HaraldWalker/user-agent-utils#115]
Example user agent - "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:40.0) Gecko/20100101 Firefox/40.0".

In version 1.14 (previous one we used) - DeviceType = Computer .
In version 1.17 - DeviceType = Unknown .

Seems like the commit that added Ubuntu and Ubuntu Mobile support in version 1.17 broke this.
Unless this was done on purpose for some reason, it looks like a bug.

I have a PR ready with a fix (though having some problems with building the project).

**Comment by HaraldWalker** [comment]
It kind of was on purpose since there are also tablet and mobile devices running on Ubuntu. Based on the user agent like that we can not be sure any longer that it is a computer. I wasn't able to find any examples of Ubuntu on tablet. So, the parent element is UNKNOWN. Am wondering if we could define a child as UBUNTU_DESKTOP. Maybe it could match on something like: Linux x86 or X11.

**Comment by itaykor** [comment]
Thanks for the quick response!
A child property would be great.
Any way I can assist with it ?

**Comment by HaraldWalker** [comment]
@itaykor do you have any more Ubuntu user agent examples? I got one example of Ubuntu on a tablet.

**Comment by itaykor** [comment]
@HaraldWalker - I have one from Chrome on my computer.
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/44.0.2403.155 Safari/537.36

I can search for examples on the internet...
From the user agent you have, can you differentiate it from the desktop ones ?

**Comment by HaraldWalker** [comment]
@itaykor, so Chrome doesn't even reveal that it is Ubuntu. I noticed that last week already. I think Firefox does.

on a tablet (submitted by a user on an ubuntu forum):
Mozilla/5.0 (Linux; Ubuntu 14.04 like Android 4.4) AppleWebKit/537.36 Chromium/35.1870.2 Mobile Safari/537.36

Unfortunately the same as mobile:
Mozilla/5.0 (Linux; Ubuntu 14.04 like Android 4.4) AppleWebKit/537.36 Chromium/35.0.1870.2 Mobile Safari/537.36

On Android it is the convention not to use Mobile Safari in the user-agent of a tablet. 
So, it seems the desktop can be differentiated from tablet/mobile and at the moment there is no good way to detect Ubuntu on a tablet unless they change the user-agent.

I've just merged your pull request. For now that's the best we can do.
